sdks/go/pkg/beam/core/runtime/graphx/xlang.go

```diff
@@ -16,11 +16,222 @@
 package graphx
 
 import (
+	"fmt"
+
 	"github.com/apache/beam/sdks/go/pkg/beam/core/graph"
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 	pipepb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
 )
 
+// MergeExpandedWithPipeline adds expanded components of all ExternalTransforms to the existing pipeline
+func MergeExpandedWithPipeline(edges []*graph.MultiEdge, p *pipepb.Pipeline) {
+	// Adding Expanded transforms to their counterparts in the Pipeline
+
+	for _, e := range edges {
+		if e.Op == graph.External {
+			exp := e.External.Expanded
+			if exp == nil {
+				continue
+			}
+			id := fmt.Sprintf("e%v", e.ID())
+
+			p.Requirements = append(p.Requirements, exp.Requirements...)
+
+			// Adding components of the Expanded Transforms to the current Pipeline
+			components, err := ExpandedComponents(exp)
+			if err != nil {
+				panic(err)
+			}
+			for k, v := range components.GetTransforms() {
+				p.Components.Transforms[k] = v
+			}
+			for k, v := range components.GetPcollections() {
+				p.Components.Pcollections[k] = v
+			}
+			for k, v := range components.GetWindowingStrategies() {
+				p.Components.WindowingStrategies[k] = v
+			}
+			for k, v := range components.GetCoders() {
+				p.Components.Coders[k] = v
+			}
+			for k, v := range components.GetEnvironments() {
+				if k == "go" {
```

Comment:
Since this is now in the graphx package, we can probably promote the constant in addDefaultEnv to a proper unexported package constant, and refer to it here.

Reply:
Makes sense. Done.